core/vm/actor/builtin/v2/verified_registry/verified_registry_actor.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vm/actor/builtin/v2/verified_registry/verified_registry_actor.hpp"
+#include "vm/actor/builtin/v0/verified_registry/verified_registry_actor_utils.hpp"
+
+namespace fc::vm::actor::builtin::v2::verified_registry {
+  namespace Utils = utils::verified_registry;
+
+  // AddVerifier
+  //============================================================================
+
+  ACTOR_METHOD_IMPL(AddVerifier) {
+    OUTCOME_TRY(Utils::checkDealSize(params.allowance));
+
+    const auto verifier = runtime.resolveAddress(params.address);
+    REQUIRE_NO_ERROR(verifier, VMExitCode::kErrIllegalState);
+
+    OUTCOME_TRY(state, runtime.getCurrentActorStateCbor<State>());
+    OUTCOME_TRY(runtime.validateImmediateCallerIs(state.root_key));
+    OUTCOME_TRY(Utils::checkAddress<State>(state, verifier.value()));
```

Comment:
Can omit `<State>`

Reply:
Yes, because v0 & v2 states are equal.
However if it is changed in the next versions, we still could use this method